src/main/java/net/snowflake/ingest/connection/ServiceResponseHandler.java

```diff
@@ -187,36 +226,53 @@ public static <T> T unmarshallStreamingIngestResponse(
    *
    * @param response HttpResponse
    * @param requestId
+   * @param apiName enum to represent the corresponding api name
+   * @param httpClient HttpClient for retries
+   * @param request HttpRequest for retries
+   * @param requestBuilder RequestBuilder for retries
+   * @return modified HttpResponse
    * @throws IOException if our entity is somehow corrupt or we can't get it
    * @throws IngestResponseException - for all other non OK status
    * @throws BackOffException - if we have a 503 issue
    */
-  private static void handleExceptionalStatus(
-      HttpResponse response, UUID requestId, ApiName apiName)
+  private static HttpResponse handleExceptionalStatus(
```

Comment:
It's kind of weird that we're passing HttpResponse as an input and return a HttpResponse as well, a better way to do it is to retry the request before calling this function and pass the final the final response here, this is just a suggestion and you don't have to change it if it involves too many changes